### PR TITLE
apr: align with Linux formula

### PR DESCRIPTION
**Pull request by SeekingMeaning**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Linux formula: https://github.com/Homebrew/linuxbrew-core/blob/master/Formula/apr.rb

### CODE REVIEW

File: Formula/apr.rb [hunk 1]
```diff
@@ -49,14 +50,37 @@ def install
     system "./configure", "--prefix=#{libexec}"
     system "make", "install"
     bin.install_symlink Dir["#{libexec}/bin/*"]
+    lib.install_symlink Dir["#{libexec}/lib/*.a"]
+    on_macos do
+      lib.install_symlink Dir["#{libexec}/lib/*.dylib"]
```
Comment (iMichka) [inline Formula/apr.rb:55]: This should work (I think we already used it somewhere else): `lib.install_symlink Dir["#{libexec}/lib/#{shared_library("*")}"]`

Comment (SeekingMeaning) [reply Formula/apr.rb:55]: It seems it only supports `*.so` files, and not `*.so.*` files: https://github.com/Homebrew/brew/blob/master/Library/Homebrew/extend/os/linux/formula.rb#L7-L9

`ipopt` uses `shared_library` and, I'm not sure but it seems to be missing some lib files, notably `libmpiseq.so.*` and `libpord.so.*`

`ipopt` bottle: https://bintray.com/linuxbrew/bottles/ipopt/3.13.3_2#files
`ipopt` formula install: https://github.com/Homebrew/homebrew-core/blob/master/Formula/ipopt.rb#L54-L58

**Comment by SeekingMeaning** [comment]
10.15:

```
brew test --retry --verbose httpd
brew test --retry --verbose php
brew test --retry --verbose php@7.3
brew test --retry --verbose php@7.4
```

File: Formula/apr.rb (hunk 1 above)
Comment (iMichka) [reply Formula/apr.rb:55]: Cool. Let's merge that and use it :)

**Review by iMichka — DISMISSED** [review]

**Comment by SeekingMeaning** [comment]
It seems `php`, `php@7.3`, `php@7.4`, and `httpd` need revision bumps, but I'm not sure why

**Comment by iMichka** [comment]
I think that it was using the static libraries before, and now we ship the shared libraries, so it links against these first?
It's weird that it failed only on 10.15.

**Comment by SeekingMeaning** [comment]
> I think that it was using the static libraries before, and now we ship the shared libraries, so it links against these first?

No, that can't be it—the logs reference `<prefix>/libexec/lib`, and all this PR does is add symlinks for `<prefix>/lib`

**Comment by iMichka** [comment]
Then I don't know. Just add the revision bumps and see if it fixes it?

**Comment by SMillerDev** [comment]
Likely it's the change in apr-config?

**Review by iMichka — APPROVED** [review]

**Pull request closed without merging**